client_test.go

```diff
@@ -30,3 +33,32 @@ func TestHostPortNoPort(t *testing.T) {
 		}
 	}
 }
+
+type InfiniteReader struct{}
+
+func (reader InfiniteReader) Read(b []byte) (int, error) {
+	for i := range b {
+		b[i] = 'A'
+	}
+	return len(b), nil
+}
+
+func BenchmarkSingleBufferedRead(b *testing.B) {
+	inf := InfiniteReader{}
+	singleReader := bufio.NewReaderSize(inf, 81920)
+	r := make([]byte, 400)
+	for n := 0; n < b.N; n++ {
+		_, _ = singleReader.Read(r)
+	}
+}
+
+func BenchmarkMultiBufferedRead(b *testing.B) {
+	inf := InfiniteReader{}
+	emptyR := new([]byte)
+	emptyRemainder := bytes.NewReader(*emptyR)
+	multiReader := bufio.NewReaderSize(io.MultiReader(emptyRemainder, bufio.NewReaderSize(inf, 81920)), 81920)
```

Comment:
Nit: for these tests, it might be helpful to phrase `81920` as the product of whatever the underlying factors are here (`5 * 16384` I think?). It's not obvious to me why these values are used here, so a comment might work too if this is some standard value I'm overlooking.

Reply:
Agreed, I should change this. It was 80 KB (1024 * 80) and 400 was just intended to be not neatly aligned with powers of 2 to trigger double buffer refill